source/extensions/tracers/skywalking/skywalking_types.cc

```diff
@@ -0,0 +1,174 @@
+#include "extensions/tracers/skywalking/skywalking_types.h"
+
+#include "envoy/common/exception.h"
+
+#include "common/common/base64.h"
+#include "common/common/empty_string.h"
+#include "common/common/fmt.h"
+#include "common/common/hex.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace SkyWalking {
+
+namespace {
+
+// The standard header name is "sw8", as mentioned in:
+// https://github.com/apache/skywalking/blob/v8.1.0/docs/en/protocols/Skywalking-Cross-Process-Propagation-Headers-Protocol-v3.md.
+const Http::LowerCaseString& propagationHeader() {
+  CONSTRUCT_ON_FIRST_USE(Http::LowerCaseString, "sw8");
+}
+
+std::string generateId(Random::RandomGenerator& random_generator) {
+  return absl::StrCat(Hex::uint64ToHex(random_generator.random()),
+                      Hex::uint64ToHex(random_generator.random()));
+}
+
+std::string base64Encode(const absl::string_view input) {
+  return Base64::encode(input.data(), input.length());
+}
+
+// Decode and validate fields of propagation header.
+std::string base64Decode(absl::string_view input) {
+  std::string result = input.length() % 4 ? EMPTY_STRING : Base64::decodeWithoutPadding(input);
```

Comment:
If we do decodeWithoutPadding, you shouldn't check length % 4? since that's what padding for.

Reply:
yep, this check is unnecessary.

Reply:
But my intention is to ensure that the input Base64 string is a string with padding.
And `decodeWithoutPadding` is different from its name. It can actually parse Base64 strings with padding.